bin/i18n/redact_restore_utils.rb

```diff
@@ -7,14 +7,24 @@
 require_relative 'i18n_script_utils'
 
 class RedactRestoreUtils
+  # RedactRestoreUtils manages the
+  def self.backup_source_file(source_path)
+    return unless File.exist?(source_path)
+    return unless source_path[I18N_SOURCE_DIR]
+    backup_path = source_path.sub(I18N_SOURCE_DIR, I18N_ORIGINAL_DIR)
+    I18nScriptUtils.copy_file(source_path, backup_path)
+  end
+
+  # For a given file, redact_file returns the redacted content of the file. The default file format is Markdown.
   def self.redact_file(source_path, plugins = [], format = 'md')
     args = [CDO.dir('bin/i18n/node_modules/.bin/redact')]
     args.push("-p #{plugins_to_arg(plugins)}") unless plugins.empty?
     args.push("-f #{format}")
     args.push(Shellwords.escape(source_path))
+
     stdout, _status = Open3.capture2(args.join(" "))
 
-    return JSON.parse(stdout)
```

Comment:
This `JSON.parse` was assuming the redacted file was in json forma.
Instead of creating a new method for redacting markdown files, I repurposed this method, and made all JSON files to be redacted through the `redact_data` method.